### ISSUE TITLE
Add documentation on all remaining models

### ISSUE THREAD

**Issue by tjsego** [tjsego/PersistentCell#7]

**Comment by tjsego** [comment]
My edits for MODEL006 and MODEL007 need confirmed/revised. Then this issue is resolved for all models in current usage.

**Comment by starboerg** [comment]
I updated model006 description here af09f28c3bf2ed41cc711d77b880f6ac47c6df00, adding implementation details as suggested by Inge. 
Please have a look.

Since this is the first multi-paradigm model we should also discuss how MCS relate to simulation time e.g. as used time continuous schemes (ODEs, Events).

**Comment by starboerg** [comment]
Following our discussion from yesterday I would like to add 2 more parameters to our model schemes:


**"cpm_motility_activation"** : ["extension", "retraction", "extension+retraction"]

 **"cpm_update_direction"** : ["source-to-target","surface-normal","cell-mass-displacement"]

- **source-to-target**: the classical voxel-based source voxel to target voxel. I assume it to be normalized to unit vector.
- **surface-normal** : Estimate the normal of the local surface using a kernel. The simplest implementation is the mean orientation of all neighbouring voxels of same stat to the target, normalized. The particular choice of the source would be irrelevant.
- **cell-mass-displacement**: product of cell displacement and cell size. Has the same order of magnitude as the other measures and corresponds to the cell-centric displacement work of the update.

I do not expect to implement all methods, but rather to document them for reproducibility purposes and to allow anyone to perform some study on it's own. All of them are reasonable choices and we can set a standard how to document them in other models.

**Comment by tjsego** [comment]
> Following our discussion from yesterday I would like to add 2 more parameters to our model schemes:
> 
> **"cpm_motility_activation"** : ["extension", "retraction", "extension+retraction"]
> 
> **"cpm_update_direction"** : ["source-to-target","surface-normal","cell-mass-displacement"]
> 
> * **source-to-target**: the classical voxel-based source voxel to target voxel. I assume it to be normalized to unit vector.
> * **surface-normal** : Estimate the normal of the local surface using a kernel. The simplest implementation is the mean orientation of all neighbouring voxels of same stat to the target, normalized. The particular choice of the source would be irrelevant.
> * **cell-mass-displacement**: product of cell displacement and cell size. Has the same order of magnitude as the other measures and corresponds to the cell-centric displacement work of the update.
> 
> I do not expect to implement all methods, but rather to document them for reproducibility purposes and to allow anyone to perform some study on it's own. All of them are reasonable choices and we can set a standard how to document them in other models.

These would be entries for models (*e.g.*, in "model_args" for relevant models), correct?

**Comment by starboerg** [comment]
I would say so. However, before I add them, I'd like to get feedback on the naming -- and their precise meaning ;-)

**Comment by ingewortel** [comment]
> I updated model006 description here [af09f28](https://github.com/tjsego/PersistentCell/commit/af09f28c3bf2ed41cc711d77b880f6ac47c6df00), adding implementation details as suggested by Inge. Please have a look.

Thanks for the update! I added minor edits/typo fixes in the comments here [af09f28](https://github.com/tjsego/PersistentCell/commit/af09f28c3bf2ed41cc711d77b880f6ac47c6df00). 

> Since this is the first multi-paradigm model we should also discuss how MCS relate to simulation time e.g. as used time continuous schemes (ODEs, Events).

Agreed (although we might also focus on first reproducing within CPM for models 003, 005, 006, before entering additional room for bugs!)

**Comment by ingewortel** [comment]
> I do not expect to implement all methods, but rather to document them for reproducibility purposes and to allow anyone to perform some study on it's own. All of them are reasonable choices and we can set a standard how to document them in other models.

> **"cpm_motility_activation"** : ["extension", "retraction", "extension+retraction"]

Agree, although perhaps this is not only up to us - we recently had this naming discussion also in the context of the chemotaxis/angiogenesis projects. I'd suggest we make a reasonable choice for now but keep in mind that we may want to update later.

As for the parameter name itself: how about "cpm_force_mode" or "cpm_force_direction"? This choice is one that will be required for basically all models with a work/force term that is defined on $\Delta H$ rather than $H$ level, so I'd suggest we look for a name that reflects that. 


> **"cpm_update_direction"** : ["source-to-target","surface-normal","cell-mass-displacement"]

naming sounds good to me!

> * **source-to-target**: the classical voxel-based source voxel to target voxel. I assume it to be normalized to unit vector.

I typically do not normalize this vector as diagonal copies act over a slighly larger distance compared to horizontal/vertical ones, and therefore, the work (force x distance) is larger. I think either choice is justifiable but I don't think we can assume either way, so then that should probably be an additional argument in our specification.

> * **surface-normal** : ... mean orientation of all neighbouring voxels of same stat to the target, normalized. The particular choice of the source would be irrelevant.

Not sure what you mean -- could you explain?

> * **cell-mass-displacement**: product of cell displacement and cell size. Has the same order of magnitude as the other measures and corresponds to the cell-centric displacement work of the update.

with this you mean $a_\sigma \vec{\delta C}$ from model006?

**Comment by ingewortel** [comment]
> Since this is the first multi-paradigm model we should also discuss how MCS relate to simulation time e.g. as used time continuous schemes (ODEs, Events).

Shall we just specify 1 MCS = 1 second for now in our model schema file? I've just created a PR with model documentation update here: https://github.com/tjsego/PersistentCell/pull/17. Since there is now a shared section for CPM definitions, we can just add a statement 1 MCS = 1 second there for now?

Other than that, I think this has been resolved other than open discussions in #14 and #15 , so if we can agree on the time relation I propse we close this issue and move the discussion there.

**Comment by starboerg** [comment]
Agree, but let's just fix the relation to continuous time and keep that as an arbitrary unit (a.t.u. -- arbitrary time unit)

**Comment by tjsego** [comment]
What are the action items left to resolve this that aren't being tracked in other issues?

**Comment by ingewortel** [comment]
I think mostly how/where exactly to specify the time relationship. I propose putting a line in models.md like so:

"In all CPM models, we interpret MCS as continuous time and specify 1 MCS = 1 a.t.u (arbitrary time unit)." 

@starboerg would that work? 

- [ ] Agree on how/where to specify time relationship
- [ ] implement

**Comment by starboerg** [comment]
Spelling should be like : We relate the time discrete Monte Carlo Step to continuous simulation time such that  1 MCS corresponds to 1 a.t.u. or just 1 simulation time. Future models may handle this differently, in particular if their simulation time has natural units. Morpheus defines and `MCSDuration` for that, since simulation time is the reference system for us.

**Comment by ingewortel** [comment]
Done in https://github.com/tjsego/PersistentCell/commit/97b3f1bfeb8abbffcf1e7400d36e3ac2ad47cb2a. @starboerg feel free to close once you are happy (or edit models.md directly if not).
